externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala

```diff
@@ -143,4 +143,21 @@ class ExecuteStatement(
       }, queryTimeout, TimeUnit.SECONDS)
     }
   }
+
+  override def cancel(): Unit = {
+    spark.sparkContext.removeSparkListener(operationListener)
```

Comment:
is it better to define a private method for this?

```
private def removeOperationListener(): Unit = {
  if (operationListener != null) {
    spark.sparkContext.removeSparkListener(operationListener)
    operationListener = null
  }
}

```

Reply:
now we want to get the state that this operation has finished by SparkListenerEvent.
this PR will be revised.